marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -320,6 +320,17 @@ function convertStepReferenceToInlineStep(stepId) {
   return newFlowStep;
 }
 
+function cleanArtifact(artifact, propertiesToReturn) {
```

Comment:
I think "clean" typically means - "remove bad/malformed stuff from it". But the intent here is to only return certain top-level properties of each step. I think "removeAllPropertiesExcept" would be a better function name.